api/src/task/notifications/generate_notifications.py

```diff
@@ -55,11 +59,39 @@ def run_task(self) -> None:
         self._send_notifications()
 
     def _collect_opportunity_notifications(self) -> None:
-        """Collect notifications for changed opportunities
-        To be implemented in future ticket
-        """
-        logger.info("Opportunity notification collection not yet implemented")
-        pass
+        """Collect notifications for changed opportunities that users are tracking"""
+        stmt = (
+            select(User.user_id, UserSavedOpportunity.opportunity_id)
+            .join(
+                UserSavedOpportunity,
+                User.user_id == UserSavedOpportunity.user_id,
```

Comment:
What if this was `select(UserSavedOpportunity)...` with similar joins. You'd avoid needing to use mappings/specific columns since the UserSavedOpportunity record is the user_id + opportunity_id.

We should make the user_notification_map opportunity IDs instead just be this class object. Might also simplify the updates later?

Reply:
Yes, you adjusted it like I was thinking